evm/src/cpu/kernel/asm/memory/core.asm

```diff
@@ -62,6 +62,24 @@
     // stack: value
 %endmacro
 
+%macro mload_kernel_code_label(label)
+    // stack: shift
+    PUSH $label  
+    ADD
+    // stack: label + shift
+    %mload_kernel_code
+    // stack: byte
+%endmacro
+
+%macro mload_kernel_code_label_u32(label)
```

Comment:
nit: I think it'd be natural to move the `%mul_const(4)` (which currently the caller does) inside this macro. So this macro would take the index of the u32, rather than its offset in bytes.